public/Aframe/samples/loading-scene-1/components.js

```diff
@@ -0,0 +1,127 @@
+        this.el.addEventListener("collide", function (e) {
+            console.log("Player has collided with ", e.detail.body.el);
+            e.detail.target.el; // Original entity (playerEl).
+            e.detail.body.el; // Other entity, which playerEl touched.
+            e.detail.contact; // Stats about the collision (CANNON.ContactEquation).
+            e.detail.contact.ni; // Normal (direction) of the collision (CANNON.Vec3).
+        });
+    },
+});
+
+AFRAME.registerComponent("detect-button", {
+    init: function () {
+        var data = this.data;
+        var el = this.el;
+
+        el.addEventListener("mouseenter", function () {
+            el.setAttribute(
+                "animation__button-enter",
+                "property: scale; to: 1.1 1.1 1.1; dur: 500; dir:alternate; startEvents: mouseenter"
+            );
+        });
+
+        el.addEventListener("mouseleave", function () {
+            el.setAttribute(
+                "animation__button-leave",
+                "property: scale; to: 1 1 1; startEvents: mouseleave"
+            );
+        });
+
+        el.addEventListener("mousedown", function () {
+            el.setAttribute(
+                "animation__button-pressed",
+                "property: object3D.position.y; to: -5.5; dir: alternate; dur: 50; startEvents: mousedown"
+            );
+
+            el.emit("buttondown", false);
+        });
+
+        el.addEventListener("mouseup", function () {
+            el.setAttribute(
+                "animation__button-released",
+                "property: object3D.position.y; to: -5; dir: alternate; dur: 100; startEvents: mouseup"
+            );
+        });
+    },
+});
+
+function addBall(x, y, z) {
+    let container = document.querySelector("#container");
+    let colors = [
+        "red",
+        "orange",
+        "yellow",
+        "green",
+        "blue",
+        "purple",
+        "hotpink",
+    ];
+
+    container.innerHTML += `<a-sphere collision-detect id="ball" click-drag dynamic-body position="${x} ${y} ${z}" radius="0.5" color="${
+        colors[Math.floor(Math.random() * colors.length)]
+    }" mass="0.5"></a-sphere>`;
+}
+
+AFRAME.registerComponent("add-ball", {
+    init: function () {
+        var data = this.data;
+        var el = this.el;
+
+        el.addEventListener("buttondown", function () {
+            var curBalls = document.querySelectorAll("#ball");
+
+            for (let i = 0; i < curBalls.length; i++) {
+                let container = document.querySelector("#container");
+                curBalls[i].parentNode.removeChild(curBalls[i]);
+            }
+
+            let x = Math.random() * 20 - 10;
+            let y = 0;
+            let z = Math.random() * -20 + 5;
+
+            for (let i = 0; i < 10; i++) {
+                addBall(x, y, z);
+            }
+        });
+    },
+});
+
+//this function doesn't work only because I'm pretty sure you somehow can't change physics in a-scene.
```

Comment:
Well, the initial gravity is 0, and you are multiplying by -1, but -1 * 0 = 0, so is that why you aren't seeing any effect of changing gravity?

Also, I recall doing this, so this is definitely doable, I think you have to do it using JS and query selector to get the <a-scene> element and using that

Reply:
It's set to 0 now, but before I was working with nonzero values without any change in result. I used my exact same approach for changing the scene's background and it worked, but for some reason, physics wasn't changing. Would you by chance have an example illustrating what you mean? I agree it doesn't make sense to me that physics would be an immutable attribute, but searching online became rather fruitless.

Reply:
You are right, I must be mistaken here. I found this issue related to it: https://github.com/n5ro/aframe-physics-system/issues/98, it does provide a workaround, but not really sure if the workaround would work for us or not. But it seems gravity can't be changed dynamically (at least not the way we are doing right now)